lib/tool_shed/util/repository_util.py

```diff
@@ -170,6 +170,81 @@ def create_repository( app, name, type, description, long_description, user_id,
     message = "Repository <b>%s</b> has been created." % escape( str( repository.name ) )
     return repository, message
 
+def update_repository( app, trans, id, **kwds):
+    """Update an existing ToolShed repository"""
+    sa_session = app.model.context.current
+    # Add the repository record to the database.
+    repository = sa_session.query( app.model.Repository ).get( app.security.decode_id( id ) )
```

Comment:
Yeah...I tried to keep it similar to how the code flow previously functioned, the line you've commented on is functionally equivalent to https://github.com/galaxyproject/galaxy/pull/30/files#diff-f4212edc0de887a9af963b0db3e6fb5cR2249 so the new code shouldn't behave any differently. If `repository` was ever `None` before, all of the same model changes/`sa_session.add` and `.flush()`es will still be executed.